python/dgl/nn/pytorch/explain/subgraphx.py

```diff
@@ -364,3 +365,453 @@ def explain_graph(self, graph, feat, target_class, **kwargs):
+                )
+
+        return mcts_node.children
+
+    def mcts_rollout(self, mcts_node):
+        r"""Perform a MCTS rollout.
+
+        Parameters
+        ----------
+        mcts_node : MCTSNode
+            Starting node for MCTS
+
+        Returns
+        -------
+        float
+            Reward for visiting the node this time
+        """
+        if (
+            sum(len(nodes) for nodes in mcts_node.nodes.values())
+            <= self.node_min
+        ):
+            return mcts_node.immediate_reward
+
+        children_nodes = self.get_mcts_children(mcts_node)
+        children_visit_sum = sum([child.num_visit for child in children_nodes])
+        children_visit_sum_sqrt = math.sqrt(children_visit_sum)
+        chosen_child = max(
+            children_nodes,
+            key=lambda c: c.total_reward / max(c.num_visit, 1)
+            + self.coef
+            * c.immediate_reward
+            * children_visit_sum_sqrt
+            / (1 + c.num_visit),
+        )
+        reward = self.mcts_rollout(chosen_child)
+        chosen_child.num_visit += 1
+        chosen_child.total_reward += reward
+
+        return reward
+
+    def explain_graph(self, graph, feat, target_class, **kwargs):
+        r"""Find the most important subgraph from the original graph for the
+        model to classify the graph into the target class.
+
+        Parameters
+        ----------
+        graph : DGLGraph
+            A heterogeneous graph
+        feat : dict[str, Tensor]
+            The dictionary that associates input node features (values) with
+            the respective node types (keys) present in the graph.
+            The input features are of shape :math:`(N_t, D_t)`. :math:`N_t` is the
+            number of nodes for node type :math:`t`, and :math:`D_t` is the feature size for
+            node type :math:`t`
+        target_class : int
+            The target class to explain
+        kwargs : dict
+            Additional arguments passed to the GNN model
+
+        Returns
+        -------
+        dict[str, Tensor]
+            The dictionary associating tensor node ids (values) to
+            node types (keys) that represents the most important subgraph
+
+        Examples
+        --------
+
+        >>> import dgl
+        >>> import dgl.function as fn
+        >>> import torch as th
+        >>> import torch.nn as nn
+        >>> import torch.nn.functional as F
+        >>> from dgl.nn import HeteroSubgraphX
+
+        >>> class Model(nn.Module):
+        ...     def __init__(self, in_dim, num_classes, canonical_etypes):
+        ...         super(Model, self).__init__()
+        ...         self.etype_weights = nn.ModuleDict(
+        ...             {
+        ...                 "_".join(c_etype): nn.Linear(in_dim, num_classes)
+        ...                 for c_etype in canonical_etypes
+        ...             }
+        ...         )
+        ...
+        ...     def forward(self, graph, feat, eweight=None):
```

Comment:
For this example, you don't need the `eweight` argument and related logic, right?